docs/paper/dev/api/command-api/arguments.mdx

```diff
@@ -0,0 +1,186 @@
+we add an enum that specifies all available values for our custom type.
+
+```java 
+public enum IceCreamType {
+    VANILLA,
+    CHOCOLATE,
+    STRAWBERRY,
+    MINT,
+    COOKIES
+}
+```
+Now, we have to define the argument itself. We do this by implementing the <Javadoc name={"io.papermc.paper.command.brigadier.argument.CustomArgumentType$Converted"}>CustomArgumentType.Converted</Javadoc> interface:
+
+```java
+public class IceCreamTypeArgument implements CustomArgumentType.Converted<IceCreamType, String> {
+
+    @Override
+    public @NotNull IceCreamType convert(String nativeType) throws CommandSyntaxException {
+        try {
+            return IceCreamType.valueOf(nativeType.toUpperCase());
+        } catch (Exception e) {
+            Message message = MessageComponentSerializer.message().serialize(Component.text("Invalid flavor %s!".formatted(nativeType), NamedTextColor.RED));
+
+            throw new CommandSyntaxException(new SimpleCommandExceptionType(message), message);
+        }
+    }
+
+    @Override
+    public @NotNull ArgumentType<String> getNativeType() {
+        return StringArgumentType.word();
+    }
+
+    @Override
+    public <S> CompletableFuture<Suggestions> listSuggestions(CommandContext<S> context, SuggestionsBuilder builder) {
+        for (IceCreamType flavor : IceCreamType.values()) {
+            builder.suggest(flavor.name(), MessageComponentSerializer.message().serialize(Component.text("look at this cool green tooltip!", NamedTextColor.GREEN)));
+        }
+
+        return CompletableFuture.completedFuture(
+                builder.build()
+        );
+    }
+}
+```
+
+That's a lot of code, so let's start from the top. We implemented the <Javadoc name={"io.papermc.paper.command.brigadier.argument.CustomArgumentType$Converted"}>CustomArgumentType.Converted</Javadoc>
+interface. This interface takes two type arguments: our custom enum, T, and a type that is native to 
+Minecraft, such as String, Integer, etc., called N. The native type exists so that the client can use
+the input data, as it doesn't know what our custom `IceCreamType` is.
+
+The first method called `convert()` converts the native type (in this case `String`)
+into our own custom type called `IceCreamType`, as the name suggests. This is so we 
+can keep using our custom type internally. If we wouldn't do that, there would be no point in using a 
+custom type. If the specified input isn't found in our enum of available flavors, we
+output an error.
+
+The second method called `getNativeType()` returns the type of string that our command argument uses.
+In our case, we're using a single word, so we return `StringArgumentType.word()`.
+
+In the last method, `listSuggestions()`, we return `CompletableFuture<Suggestions>` so that the client
+can suggest all available options. We can even add tooltips to the suggestions to explain them in greater
+detail. 
+
+As a last step, we need to register the Command:
```

Comment:
```suggestion
We implemented the <Javadoc name={"io.papermc.paper.command.brigadier.argument.CustomArgumentType$Converted"}>CustomArgumentType.Converted</Javadoc>
interface. This interface takes two type arguments: our custom enum, T, and a native Java type called N.

- `convert()` converts the native type (in this case `String`) into our custom type.

-  `getNativeType()` returns the type of string that our command argument uses. This uses a single word, so we return `StringArgumentType.word()`.

- `listSuggestions()` returns `CompletableFuture<Suggestions>` so that the client
can suggest all available options. We can even add tooltips to the suggestions to explain them in greater
detail. 

We then need to register the command:
```

so much of this was unnecessary waffle to me